source/shared/login-type-vo.js

```diff
@@ -0,0 +1,63 @@
+LoginType = Space.domain.ValueObject.extend('LoginType', {
+
+  Constructor(loginTypeParam) {
+
+    let loginType = undefined;
+
+    // Allow to provide another instance of LoginType as param
+    if (loginTypeParam instanceof Object) {
+      loginType = loginTypeParam.loginType;
+    } else {
+      loginType = loginTypeParam;
+    }
+
+    if (!LoginType.isValid(loginType)) {
+      throw new Error(LoginType.ERRORS.invalidloginType(loginType));
+    }
+
+    this.loginType = loginType;
+    Object.freeze(this);
+
+  },
+
+  toString() {
+    return this.loginType;
+  },
+
+  // EJSON serializable fields
+  fields() {
+    return {
+      loginType: String
+    };
+  },
+
+  isService() {
+    if (this.loginType === 'username' || this.loginType === 'email') {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+});
+
+// Register as EJSON type
+LoginType.type('LoginType');
+
+LoginType.ERRORS = {
+  invalidloginType(loginType) {
+    return `Invalid loginType '${loginType}' given.`;
+  }
+};
+
+LoginType.VALID_LOGIN_TYPES = [
```

Comment:
From perspective of a *Space package user*:
> Today I require to add Twitter support to accounts package by *myself*

Oh so I add new `ServiceConfiguration.configurations` with necessary settings and... Now I can't do anything else. You should be able to register new login handlers(?) so this package is extensible for anyone. This https://github.com/meteor-space/accounts/blob/feature/login-with-google/source/server/services/user-creation-service.js#L15 whole `if else if` should be handled via _container_ of sorts with allowed values and thats when `LoginType` does not need allowed values.